crates/chain/src/tx_graph.rs

```diff
@@ -104,36 +160,48 @@ impl TxGraph {
     }
 
     /// Iterate over all full transactions in the graph.
-    pub fn full_transactions(&self) -> impl Iterator<Item = &Transaction> {
-        self.txs.iter().filter_map(|(_, tx)| match tx {
-            TxNode::Whole(tx) => Some(tx),
-            TxNode::Partial(_) => None,
-        })
+    pub fn full_transactions(&self) -> impl Iterator<Item = TxInGraph<'_, Transaction, A>> {
+        self.txs
+            .iter()
+            .filter_map(|(&txid, (tx, anchors, last_seen))| match tx {
+                TxNode::Whole(tx) => Some(TxInGraph {
+                    txid,
+                    tx,
+                    anchors,
+                    last_seen: *last_seen,
+                }),
+                TxNode::Partial(_) => None,
+            })
     }
 
     /// Get a transaction by txid. This only returns `Some` for full transactions.
     ///
     /// Refer to [`get_txout`] for getting a specific [`TxOut`].
     ///
     /// [`get_txout`]: Self::get_txout
-    pub fn get_tx(&self, txid: Txid) -> Option<&Transaction> {
-        match self.txs.get(&txid)? {
-            TxNode::Whole(tx) => Some(tx),
-            TxNode::Partial(_) => None,
+    pub fn get_tx(&self, txid: Txid) -> Option<TxInGraph<'_, Transaction, A>> {
```

Comment:
I don't think we really needed to change the API here.
You can expose the anchors and last seen in other methods.

Reply:
Reverted in 67a4abae85b1bdaf16092a753f5f2a5f8726d769